pkg/kubectl/cmd/clusterinfo_dump.go

```diff
@@ -116,16 +112,30 @@ func setupOutputWriter(dir string, defaultWriter io.Writer, filename string) io.
 	return file
 }
 
-func (o *ClusterInfoDumpOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
-	printer, err := o.PrintFlags.ToPrinter()
-	if err != nil {
+func printJson(obj runtime.Object, w io.Writer) error {
```

Comment:
We should be using our printing stack, not circumventing it via special-cases in individual commands.
The overall problem appears to be that this command is sending internal objects to the printer.
These objects do not have a defined GroupVersionKind, so the printer refuses to print them.

To allow this command to work with the printer, we can re-work this command to work with external versions of objects: https://gist.github.com/juanvallejo/031d20ab40488c98c81b9928c5a5e753 (make sure you are up to date with latest master before applying)